store/src/java/com/zimbra/cs/service/account/ResetPassword.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * ***** BEGIN LICENSE BLOCK *****
+ * Zimbra Collaboration Suite Server
+ * Copyright (C) 2018 Synacor, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software Foundation,
+ * version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ * ***** END LICENSE BLOCK *****
+ */
+
+package com.zimbra.cs.service.account;
+
+import java.util.Map;
+
+import com.zimbra.common.account.Key.AccountBy;
+import com.zimbra.common.service.ServiceException;
+import com.zimbra.common.soap.AccountConstants;
+import com.zimbra.common.soap.Element;
+import com.zimbra.common.util.ZimbraLog;
+import com.zimbra.cs.account.Account;
+import com.zimbra.cs.account.AccountServiceException.AuthFailedServiceException;
+import com.zimbra.cs.account.AuthToken;
+import com.zimbra.cs.account.Provisioning;
+import com.zimbra.cs.service.AuthProvider;
+import com.zimbra.soap.JaxbUtil;
+import com.zimbra.soap.ZimbraSoapContext;
+import com.zimbra.soap.account.message.ResetPasswordRequest;
+
+public class ResetPassword extends AccountDocumentHandler {
+
+    public Element handle(Element request, Map<String, Object> context) throws ServiceException {
+
+        if (!checkPasswordSecurity(context)) {
+            throw ServiceException.INVALID_REQUEST("clear text password is not allowed", null);
+        }
+
+        ZimbraSoapContext zsc = getZimbraSoapContext(context);
+        ResetPasswordRequest req = JaxbUtil.elementToJaxb(request);
+        req.validateResetPasswordRequest();
+
+        String email = req.getEmail();
+        String newPassword = req.getPassword();
+
+        Provisioning prov = Provisioning.getInstance();
+        Account acct = prov.get(AccountBy.name, email, zsc.getAuthToken());
```

Comment:
Need some tests to make sure that one user can't reset the password for another user and that only people with suitable caps (user themself/admin/domain admin) can reset it.

Reply:
password is reset on basis of auth token only. so no one can change password for others.